core/trino-main/src/main/java/io/trino/sql/planner/NodePartitioningManager.java

```diff
@@ -104,7 +104,7 @@ public BucketFunction getBucketFunction(Session session, PartitioningHandle part
                 .orElseThrow(() -> new IllegalArgumentException("No connector ID for partitioning handle: " + partitioningHandle));
         ConnectorNodePartitioningProvider partitioningProvider = getPartitioningProvider(catalogName);
         BucketFunction bucketFunction = partitioningProvider.getBucketFunction(
-                partitioningHandle.getTransactionHandle().orElse(null),
+                partitioningHandle.getTransactionHandle().orElseThrow(() -> new IllegalArgumentException("No transactionHandle for partitioning handle: " + partitioningHandle)),
```

Comment:
why is transaction handle optional then? What is the case when it is not set? Worth a comment or more descriptive exception message?

Reply:
I think the empty transaction handle is for `SystemPartitioningHandle` (internal use)